keanu-python/keanu/algorithm/sampling.py

```diff
@@ -20,3 +20,17 @@ def sample(net, sample_from, algo='metropolis', draws=500, drop=0, down_sample_i
     vertex_samples = {Vertex._get_python_id(vertex_unwrapped): list(map(Tensor._to_ndarray, network_samples.get(vertex_unwrapped).asList())) for vertex_unwrapped in vertices_unwrapped}
 
     return vertex_samples
+
+def iterate_samples(net, sample_from, algo='metropolis', draws=500, drop=0, down_sample_interval=1):
+	vertices_unwrapped = k.to_java_object_list(sample_from)
+
+	network_samples = algorithms[algo].withDefaultConfig().generatePosteriorSamples(net.unwrap(), vertices_unwrapped).dropCount(drop).downSampleInterval(down_sample_interval).stream().iterator()
+	
+	return iterator_samples(network_samples, draws, vertices_unwrapped)
+
+
+def iterator_samples(network_samples, draws, vertices_unwrapped):
```

Comment:
Do we need to limit the number of draws?  Given we're streaming, I don't see a need to put an upper bound on it - instead, just allow them to keep streaming until they're done (ie they could use itertools.islice() etc).

Reply:
Sorry I've changed this to a private method so it's clearer - this function creates the generator and is returned inside the `iterate_samples` function. I need the number of draws so I know how many times I'm able to call `.next()` on my Java iterator before I run out of samples!